src/dotnet/Chat.UI.Blazor/Services/ChatUI.StateSync.cs

```diff
@@ -40,6 +41,56 @@ private async Task InvalidateSelectedChatDependencies(CancellationToken cancella
         }
     }
 
+    private async Task MaintainChatPositions(CancellationToken cancellationToken)
```

Comment:
The name is misleading. `InvalidateChatPositions`?

Reply:
Maybe `InvalidateGetChatIdByPosition`?

Reply:
IMO you shouldn't break naming conventions other ppl use - see how other methods are named in such classes, `InvalidateXxx` is a very common pattern.

Reply:
It's not just invalidating! It updates the list as well